XenAdmin/SettingsPanels/PerfmonAlertOptionsPage.cs

```diff
@@ -145,7 +157,8 @@ public bool HasChanged
                 return ((_OrigEmailNotificationCheckBox != EmailNotificationCheckBox.Checked) ||
                         (_OrigEmailAddressTextBox != EmailAddressTextBox.Text) ||
                         (_OrigSmtpServerAddrTextBox != SmtpServerAddrTextBox.Text) ||
-                        (_OrigSmtpServerPortTextBox != SmtpServerPortTextBox.Text));
+                        (_OrigSmtpServerPortTextBox != SmtpServerPortTextBox.Text) ||
+                        (_OrigMailLanguageCodeComboBox != MailLanguageComboBox.SelectedValue.ToString()));

```

Comment:
`_OrigMailLanguageCodeComboBox != MailLanguageComboBox.SelectedValue.ToString()` always returns true, because _OrigMailLanguageCodeComboBox contains the KeyValuePair (e.g. "[en-US, English]") and SelectedValue contains just the language code (e.g. "en-US")